open-reac/src/main/java/com/powsybl/openreac/OpenReacTool.java

```diff
@@ -0,0 +1,250 @@
+/**
+ * Copyright (c) 2023, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.powsybl.openreac;
+
+import com.google.auto.service.AutoService;
+import com.powsybl.ampl.converter.AmplExporter;
+import com.powsybl.commons.PowsyblException;
+import com.powsybl.commons.datasource.DataSource;
+import com.powsybl.commons.datasource.FileDataSource;
+import com.powsybl.iidm.network.Exporter;
+import com.powsybl.iidm.network.ImportConfig;
+import com.powsybl.iidm.network.Network;
+import com.powsybl.iidm.network.tools.ConversionToolUtils;
+import com.powsybl.loadflow.LoadFlow;
+import com.powsybl.loadflow.LoadFlowParameters;
+import com.powsybl.loadflow.LoadFlowResult;
+import com.powsybl.openreac.exceptions.InvalidParametersException;
+import com.powsybl.openreac.parameters.input.OpenReacParameters;
+import com.powsybl.openreac.parameters.input.algo.OpenReacAlgoParam;
+import com.powsybl.openreac.parameters.input.algo.OpenReacOptimisationObjective;
+import com.powsybl.openreac.parameters.input.algo.OptimisationVoltageRatio;
+import com.powsybl.openreac.parameters.output.OpenReacResult;
+import com.powsybl.openreac.parameters.output.ReactiveSlackOutput;
+import com.powsybl.tools.Command;
+import com.powsybl.tools.Tool;
+import com.powsybl.tools.ToolRunningContext;
+import org.apache.commons.cli.CommandLine;
+import org.apache.commons.cli.Option;
+import org.apache.commons.cli.Options;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.*;
+import java.util.stream.Collectors;
+
+import static com.powsybl.iidm.network.tools.ConversionToolUtils.*;
+
+/**
+ * @author Nicolas Pierre <nicolas.pierre at artelys.com>
+ */
+@AutoService(Tool.class)
+public class OpenReacTool implements Tool {
```

Comment:
Note for later: we need to unit test the tool

Reply:
Did a simple test on the `OpenReacParameters` parsing and on the CommandLine parsing.